### ISSUE TITLE
scaling problem for non-square res images

### ISSUE THREAD

**Issue by onelson** [ggez/ggez#65]
Given a 1024x768 image, I would expect a program with window dimensions matching the image, which renders the image at `Point::new(512., 384.)` , to look roughly identical to the source.

Here's what I put in:
![Source image](http://i.imgur.com/D8BOcRP.jpg)

Here's what came out:
![Rendered result](http://i.imgur.com/jEajjcN.png)

When I encountered this, my quick fix was to add a blank 128px to the top and bottom of the image to make it 1024x1024 (which resulted in no scaling).

I wrote this as an example, but wasn't sure if you'd want that as a PR or what since it's just for debugging the issue, so here's the source as a code block instead:

```rust
extern crate ggez;

use std::time::Duration;
use ggez::conf;
use ggez::event;
use ggez::{GameResult, Context};
use ggez::graphics;
use ggez::timer;

struct MainState {
    image: graphics::Image,
}

impl MainState {
    fn new(ctx: &mut Context) -> GameResult<MainState> {
        ctx.print_resource_stats();
        let image = graphics::Image::new(ctx, "checkers.jpg").unwrap();
        let s = MainState { image: image };
        Ok(s)
    }
}

impl event::EventHandler for MainState {
    fn update(&mut self, _ctx: &mut Context, _dt: Duration) -> GameResult<()> { Ok(()) }
    fn draw(&mut self, ctx: &mut Context) -> GameResult<()> {
        graphics::clear(ctx);
        let dest_point = graphics::Point::new(512., 384.);
        graphics::draw(ctx, &mut self.image, dest_point, 0.0)?;
        graphics::present(ctx);
        timer::sleep_until_next_frame(ctx, 60);
        Ok(())
    }
}

pub fn main() {
    let mut c = conf::Conf::new();
    c.window_width = 1024;
    c.window_height = 768;
    let ctx = &mut Context::load_from_conf("ratio-test", "ggez", c).unwrap();
    let state = &mut MainState::new(ctx).unwrap();
    if let Err(e) = event::run(ctx, state) {
        println!("Error encountered: {}", e);
    } else {
        println!("Game exited cleanly.");
    }
}
```

**Comment by onelson** [comment]
I updated the title since, well `768 as usize` itself does not pass the `.is_power_of_two()` test, and that's fine. I'd still expect the output to be in the neighborhood of the source.

**Comment by icefoxen** [comment]
Aha, thanks for the reproduction.  Non-power-of-two image stuff is still not very well tested yet I'm afraid, so I'll see what's up.  When you go from the actual image coordinates, to the rescaled image coordinates with UV's, to a rect with UV's, to a scaled rect with UV's...  There's lots of room for mix-ups to happen.

**Comment by icefoxen** [comment]
Confirmed, it's a bug in the np2 scaling... which, per kvark, should be unnecessary, so maybe we can just get rid of it entirely?  I can only hope.  See issue #59 for tracking.

**Comment by icefoxen** [comment]
Closed along with issue #59
